src/domain/settings/taxes/details.tsx

```diff
@@ -0,0 +1,216 @@
+import {
+  useAdminRegion,
+  useAdminUpdateRegion,
+  useAdminTaxRates,
+  useAdminDeleteTaxRate,
+} from "medusa-react"
+import clsx from "clsx"
+import React, { useEffect, useState } from "react"
+import { useTable } from "react-table"
+import Spinner from "../../../components/atoms/spinner"
+import PlusIcon from "../../../components/fundamentals/icons/plus-icon"
+import TrashIcon from "../../../components/fundamentals/icons/trash-icon"
+import EditIcon from "../../../components/fundamentals/icons/edit-icon"
+import Table from "../../../components/molecules/table"
+import BodyCard from "../../../components/organisms/body-card"
+import DeletePrompt from "../../../components/organisms/delete-prompt"
+import useNotification from "../../../hooks/use-notification"
+import useTaxRateColumns from "./use-tax-rate-columns"
+import NewTaxRate from "./new"
+import EditTaxRate from "./edit"
+import { RegionTaxForm } from "./region-form"
+import { getErrorMessage } from "../../../utils/error-messages"
+
+enum TaxRateType {
+  REGION = "region",
+  RATE = "rate",
+}
+
+type PaginationProps = {
+  limit: number
+  offset: number
+}
+
+type TaxRate = {
+  id: string
+  name?: string
+  rate: number | null
+  code: string | null
+  type: TaxRateType
+}
+
+const DEFAULT_PAGESIZE = 10
+
+const TaxDetails = ({ id, onDelete, handleSelect }) => {
+  if (!id) {
+    return null
+  }
+
+  const [pagination, setPagination] = useState<PaginationProps>({
+    limit: DEFAULT_PAGESIZE,
+    offset: 0,
+  })
+  const [showNew, setShowNew] = useState<boolean>(false)
+  const [deleteRate, setDeleteRate] = useState<TaxRate | null>(null)
+  const [editRate, setEditRate] = useState<TaxRate | null>(null)
+  const [tableEntries, setTableEntries] = useState<TaxRate[]>([])
+  const notification = useNotification()
+
+  const deleteTaxRate = useAdminDeleteTaxRate(deleteRate?.id)
+
+  const { tax_rates, isLoading: taxRatesLoading } = useAdminTaxRates({
+    region_id: id,
+    ...pagination,
+  })
+
+  const handleDelete = async () => {
+    if (!deleteRate || deleteRate.type !== TaxRateType.RATE) {
+      return Promise.resolve()
+    }
+
+    return deleteTaxRate.mutateAsync()
+  }
```

Comment:
This is likely because the db you are working with has the old tax migration applied - I have updated the migration to do cascading deletes on `ProductTaxRate + ShippingTaxRate + ProductTypeTaxRate`. You should be able to get your DB up to speed by doing the following:

### In PgAdmin
**Rollback the old migration** 
```sql
ALTER TABLE "shipping_tax_rate" DROP CONSTRAINT "FK_346e0016cf045b9980747747645";
ALTER TABLE "shipping_tax_rate" DROP CONSTRAINT "FK_f672727ab020df6c50fb64c1a70";
ALTER TABLE "product_type_tax_rate" DROP CONSTRAINT "FK_ece65a774192b34253abc4cd672";
ALTER TABLE "product_type_tax_rate" DROP CONSTRAINT "FK_25a3138bb236f63d9bb6c8ff111";
ALTER TABLE "product_tax_rate" DROP CONSTRAINT "FK_2484cf14c437a04586b07e7dddb";
ALTER TABLE "product_tax_rate" DROP CONSTRAINT "FK_1d04aebeabb6a89f87e536a124d";
ALTER TABLE "shipping_method_tax_line" DROP CONSTRAINT "FK_926ca9f29014af8091722dede08";
ALTER TABLE "line_item_tax_line" DROP CONSTRAINT "FK_5077fa54b0d037e984385dfe8ad";
ALTER TABLE "region" DROP CONSTRAINT "FK_91f88052197680f9790272aaf5b";
ALTER TABLE "tax_rate" DROP CONSTRAINT "FK_b95a1e03b051993d208366cb960";
DROP INDEX "public"."IDX_f672727ab020df6c50fb64c1a7";
DROP INDEX "public"."IDX_346e0016cf045b998074774764";
DROP INDEX "public"."IDX_25a3138bb236f63d9bb6c8ff11";
DROP INDEX "public"."IDX_ece65a774192b34253abc4cd67";
DROP INDEX "public"."IDX_1d04aebeabb6a89f87e536a124";
DROP INDEX "public"."IDX_2484cf14c437a04586b07e7ddd";
ALTER TABLE "line_item" DROP COLUMN "is_return";
ALTER TABLE "region" DROP COLUMN "tax_provider_id";
ALTER TABLE "region" DROP COLUMN "automatic_taxes";
ALTER TABLE "region" DROP COLUMN "gift_cards_taxable";
DROP TABLE "shipping_tax_rate";
DROP TABLE "product_type_tax_rate";
DROP TABLE "product_tax_rate";
DROP INDEX "public"."IDX_926ca9f29014af8091722dede0";
DROP TABLE "shipping_method_tax_line";
DROP INDEX "public"."IDX_5077fa54b0d037e984385dfe8a";
DROP TABLE "line_item_tax_line";
DROP TABLE "tax_provider";
DROP TABLE "tax_rate";
```

**Delete tax system migration in migrations table**
```sql
delete from migrations where name = 'newTaxSystem1641636508055';
```

### In dev project
Make sure that you have the latest version of `tax-api` installed.

There is a canary version so you should be able to do: `yarn add @medusajs/medusa@tax-api`

```
medusa migrations run
```

### Test again 🎉

Reply:
Works like a charm! 🥳